spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -195,4 +196,29 @@
 	 * @return the final result of the transaction.
 	 */
 	<A> A performTransaction(Function<DatastoreOperations, A> operations);
+
+	/**
+	 * Get a Datastore entity based on a id and convert it to a map.
+	 * @param key the key of the entity
+	 * @param valueType type values should be converted to
+	 * @return if an entity for a given key exists, returns the map representation of it,
+	 * {@code null} otherwise
+	 */
+	<T> Map<String, T> findByIdAsMap(Key key, Class<T> valueType);
+
+	/**
+	 * Save a map as a Datastore entity
+	 * @param key the key for the entity
+	 * @param map a map
```

Comment:
I think we can do better for javadoc.
Maybe something describing the fact that the map keys will be used as column names?